File: src/main/java/tech/zolhungaj/amqapi/servercommands/FriendAdded.java
package tech.zolhungaj.amqapi.servercommands;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.objects.PlayerGameState;

public record FriendAdded(
        @Json(name = "name") String playerName,
        Boolean online,
        Integer status,
        PlayerGameState gameState
) implements Command{
    @Override
    public String getCommandName() {
        return CommandType.NEW_FRIEND.commandName;
    }
}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/social/FriendAdded.kt
package tech.zolhungaj.amqapi.servercommands.social

import com.squareup.moshi.Json
import tech.zolhungaj.amqapi.servercommands.CommandType
import tech.zolhungaj.amqapi.servercommands.objects.PlayerGameState
import tech.zolhungaj.amqapi.servercommands.objects.PlayerStatus

@JvmRecord
@CommandType("new friend")
data class FriendAdded(
    @Json(name = "name") val playerName: String,
    val online: Boolean,
    val status: PlayerStatus,
    val gameState: PlayerGameState?
)
